packages/amplify-cli/src/init-steps/preInitSetup.ts

```diff
@@ -22,6 +35,38 @@ export const preInitSetup = async (context: $TSContext): Promise<$TSContext> =>
   return context;
 };
 
+/**
+ * recommend using Gen 2 or continue with Gen 1.
+ * ask for why they are using Gen 1 and store the answer in project-config
+ */
+const gen2Recommendation = async (context: $TSContext): Promise<$TSContext> => {
+  if (!isNewProject(context)) {
+    return context;
+  }
+  printer.warn(
+    `AWS Amplify recommends using Amplify Gen 2 for new projects. Learn how to get started at https://docs.amplify.aws/react/start/quickstart/`,
+  );
+
+  const continueWithGen1 = await prompter.confirmContinue('Do you want to continue with Amplify Gen 1?');
+
+  if (!continueWithGen1) {
+    process.exit(0);
+  }
+
+  const whyContinueWithGen1 = await prompter.pick('Why would you like to use Amplify Gen 1?', [
+    'I am a current Gen 1 user',
+    'Gen 2 is missing features I need from Gen 1',
+    'I find the Gen 1 CLI easier to use',
+    'Prefer not to answer',
+  ]);
+
+  context.exeInfo.projectConfig = {
```

Comment:
does `context.exeInfo.projectConfig` already have any content at this point? If so, this line would erase that

Reply:
It does not yet, because it's the `preInitSetup`. 

In the same PR in the s0-analyzeProject.ts file, it was where `projectConfig` was first generated, so `...context.exeInfo.projectConfig,` is added to prevent overriden.

Reply:
Yes. Before this PR, it was first added in s0-analyzeProject.ts line 154 in this PR.

Reply:
Looks like that file might blow this away then: https://github.com/aws-amplify/amplify-cli/blob/2570d0c448b63e6cad419757030fd9f26eb97557/packages/amplify-cli/src/init-steps/s0-analyzeProject.ts#L154

Where is this value being consumed? Do we have tests around that?

Reply:
It won't be blown away because it's already taken cared in this PR

https://github.com/aws-amplify/amplify-cli/pull/13849/files#diff-472b813f47ea0bbcba3522fb7e9f5ae4633dc0f14d44b8a074370baa67e5d961R155

We don't have integ test for the stack metadata yet...